### PR TITLE
Adding Mix Release with Docker

### PR DESCRIPTION
**Pull request by andyduong1920**
Solve https://github.com/nimblehq/elixir-templates/issues/12

## What happened

Dockerize the App with Mix Release
 
## Insight

- Update `config/prod.exs`
- Create the `config/runtime.exs`
- Adding `Dockerfile, docker-compose.yml`
- Adding the `bin/start.sh` for the Docker entry point, I decided to put that in the `bin/` folder as because after the app is built, it has the `bin/` folder there.

image

Refer: [Phoenix release](https://hexdocs.pm/phoenix/releases.html)

- Adding Docker build script on Github workflow to test the Build image and app scripts

## Proof Of Work

https://drive.google.com/file/d/1rsOeWJBmAvvIC6uej9JTkpQFoSM51KJf/view

### CODE REVIEW

File: .credo.exs [hunk 1]
```diff
@@ -170,7 +170,7 @@
         {Credo.Check.Refactor.DoubleBooleanNegation, []},
         {Credo.Check.Refactor.ModuleDependencies, false},
         {Credo.Check.Refactor.NegatedIsNil, []},
-        {Credo.Check.Refactor.PipeChainStart, []},
+        {Credo.Check.Refactor.PipeChainStart, false},
```
Comment (olivierobert) [inline .credo.exs:173]: Is this really needed as [per the example](https://github.com/rrrene/credo/blob/master/lib/credo/check/refactor/pipe_chain_start.ex), this is actually nicer to read

File: priv/templates/nimble.phx.gen.template/Dockerfile.eex [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+# prepare release image
+FROM alpine:3.12.0 AS app
+RUN apk add --no-cache openssl ncurses-libs
+
+WORKDIR /app
+
+RUN chown nobody:nobody /app
+
+USER nobody:nobody
```
Comment (olivierobert) [inline priv/templates/nimble.phx.gen.template/Dockerfile.eex:42]: Not sure if `nobody` is a standard non-root user name and group but maybe having something more descriptive would be better

**Review by olivierobert — COMMENTED** [review]
Looks good

File: .credo.exs (hunk 1 above)
Comment (andyduong1920) [reply .credo.exs]: Reverted in https://github.com/nimblehq/elixir-templates/pull/37/commits/8ef2ac4d2dfcb4ac4f9bd1a657aea819c847fdb2

File: priv/templates/nimble.phx.gen.template/Dockerfile.eex (hunk 2 above)
Comment (andyduong1920) [reply priv/templates/nimble.phx.gen.template/Dockerfile.eex]: changed it to `app_group` and `app_user` @olivierobert

**Review by olivierobert — APPROVED** [review]

**Review by hanam1ni — APPROVED** [review]

**Pull request merged**